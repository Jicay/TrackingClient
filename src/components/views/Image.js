import React, {Component} from "react";
import {Link} from "react-router-dom";

export default class Image extends Component {

    constructor(props) {
        super(props);
    }

    render() {
        const {image} = this.props;

        const pathVariable = image.replace("/", "-");

        return (
            <div className='col align-content-center'>
                <Link to={'/images/' + pathVariable}>
                    <img className="img-fluid" src={'https://live.staticflickr.com/' + image + '.jpg'}/>
                </Link>
            </div>
        )
    }


}