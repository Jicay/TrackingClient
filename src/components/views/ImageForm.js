import React, {Component} from "react";
import {Container,} from 'reactstrap';
import Image from "./Image";
import EmailForm from "./EmailForm";

export default class ImageForm extends Component {
    constructor(props) {
        super(props);
    }

    componentDidMount() {
        const {onLoad} = this.props;
        onLoad();
    }

    render() {
        const {history, match} = this.props;
        const imageUrl = match.params.imageUrl.replace('-', '/');

        return (
            <Container className="image container" id="page" >
                <div className='row'>
                    <Image image={imageUrl}/>
                </div>
                <div className='row'>
                    <EmailForm history={history}/>
                </div>
            </Container>
        );
    }
}