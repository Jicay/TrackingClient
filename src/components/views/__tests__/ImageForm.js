import React from 'react';
import renderer from 'react-test-renderer';
import {MemoryRouter} from "react-router";
import Image from "../Image";
import ImageForm from "../ImageForm";

it('renders correctly', () => {
    const history = {
        location: '/image'
    };

    const match = {
        params: {
            imageUrl: '45645-qddfsfd8754fs'
        }
    };

    const tree = renderer
        .create(<MemoryRouter><ImageForm history={history} match={match} onLoad={jest.fn()}/></MemoryRouter>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});