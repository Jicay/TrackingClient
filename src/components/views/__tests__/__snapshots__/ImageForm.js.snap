// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="image container container"
  id="page"
>
  <div
    className="row"
  >
    <div
      className="col align-content-center"
    >
      <a
        href="/images/45645-qddfsfd8754fs"
        onClick={[Function]}
      >
        <img
          className="img-fluid"
          src="https://live.staticflickr.com/45645/qddfsfd8754fs.jpg"
        />
      </a>
    </div>
  </div>
  <div
    className="row"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="inputMail"
        >
          Email :
        </label>
        <input
          id="inputMail"
          onChange={[Function]}
          placeholder="Saisissez votre adresse e-mail"
          type="email"
          value=""
        />
      </div>
      <button
        className="btn btn-primary"
        type="submit"
      >
        Envoyer
      </button>
    </form>
  </div>
</div>
`;
